### PR TITLE
refactor(button/icon-button/button-group): style updates

### PR DESCRIPTION
**Pull request by SisIvanova**
Related to IgniteUI/igniteui-webcomponents/issues/1107

[Button Handoff](https://www.figma.com/file/2dnNPqpR7K7tMrWvf3jHpO/Button?type=design&node-id=116-22639&mode=design&t=xsh6UPZPQLvNJN9y-0)

[Button Group Handoff](https://www.figma.com/file/0iCinYLqmAN5sbv305sw0C/branch/Q4Q0fIhkpb02wjrNYwCVTc/Button-Group?type=design&node-id=101-7876&mode=design&t=taVAOZ773Zkhgcw0-0)

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by SisIvanova** [comment]
@didimmova @AnjiManova Please verify if the width of the buttons in all types is correct.

**Comment by didimmova** [comment]
In Bootstrap button group a button with focus-visible + active has foreground of primary-800, I think it should be white. It's triggered with the keyboard by navigating with tab to the desired button and clicking space on it.

Screenshot 2024-04-19 at 15 23 51

**Review by didimmova — DISMISSED** [review]

**Comment by AnjiManova** [comment]
> @didimmova @AnjiManova Please verify if the width of the buttons in all types is correct.

The widths of all button types look fine to me. (No matter the type, if the content is the same, the widths are equal.)

**Comment by AnjiManova** [comment]
**Fluent**
I know that the focus in this PR is not Fluent, but it was mentioned that some changes are made there as well, so I checked it

Button Group
https://github.com/IgniteUI/igniteui-angular/assets/127843837/662996be-a147-4ebf-b2c7-1c0edfd43af4
as we discussed the focused border shouldn't stay after interaction with the button. Applicable for Light and Dark mode.

**Comment by imincheva** [comment]
**Button Group**
- Material: Min-width of the button group button should be 50px;
- Bootstrap: Min-width of the button group button should be 42px;

**Comment by AnjiManova** [comment]
All kits have been reviewed, and there are no comments. Every comment is implemented.

**Comment by sbayreva** [comment]
### Material 
1. Focused and Hover state of the contained button in Light and Dark Mode - the elevation should be 8

**Comment by SisIvanova** [comment]
> ### Material
> 1. Focused and Hover state of the contained button in Light and Dark Mode - the elevation should be 8

Screenshot 2024-04-30 at 10 02 18 AM
Screenshot 2024-04-30 at 10 02 54 AM

The elevation is 8.

File: projects/igniteui-angular/src/lib/directives/button/button-base.ts [hunk 1]
```diff
@@ -35,8 +36,31 @@ export abstract class IgxButtonBaseDirective extends DisplayDensityBase {
...
+     * Default value is `false`.
+     * ```typescript
+     * this.button.focus = true;
+     * ```
+     * ```typescript
+     * let isFocused =  this.button.focused;
+     * ```
+     */
+    @HostBinding('class.igx-button--focused')
+    public focused = false;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/directives/button/button-base.ts:62]: Do we need this to be public?

File: projects/igniteui-angular/src/lib/directives/button/button-base.ts [hunk 2]
```diff
@@ -66,6 +90,22 @@ export abstract class IgxButtonBaseDirective extends DisplayDensityBase {
         super(_displayDensityOptions, element);
     }
 
+    public ngAfterViewInit() {
+        fromEvent(this.nativeElement, 'keyup').subscribe((event: KeyboardEvent) => {
+            this.updateOnKeyUp(event)
+        });
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/directives/button/button-base.ts:96]: What is the issue of going with a simple `@HostListener`?

File: projects/igniteui-angular/src/lib/directives/button/button-base.ts [hunk 3]
```diff
@@ -66,6 +90,22 @@ export abstract class IgxButtonBaseDirective extends DisplayDensityBase {
...
+    }
+
+    /**
+     * @hidden
+     * @internal
+     */
+    protected updateOnKeyUp(event: KeyboardEvent) {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/directives/button/button-base.ts:103]:
```suggestion
    @HostListener('keyup', ['$event'])
    protected updateOnKeyUp(event: KeyboardEvent) {
```

**Pull request merged**